### PR TITLE
chore: add metrics for native chunk

### LINKED ISSUE #8760: metrics: missing `Source Throughput(rows) Per Partition` for `nexmark` sources
Seems this metric is recorded separately for each kind of connector and not in a common path.

https://github.com/risingwavelabs/risingwave/blob/1c6b2465e6848deddfb4490e09de757df00defc2/src/connector/src/macros.rs#L270-L277

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

resolve #8760 

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
~~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/connector/src/source/datagen/source/reader.rs [hunk 1]
```diff
@@ -151,8 +151,24 @@ impl SplitReader for DatagenSplitReader {
...
-                spawn_data_generation_stream(self.generator.into_native_stream(), BUFFER_SIZE)
-                    .boxed()
+                let actor_id = self.source_ctx.source_info.actor_id.to_string();
+                let source_id = self.source_ctx.source_info.source_id.to_string();
+                let split_id = self.split_id.to_string();
+                let metrics = self.source_ctx.metrics.clone();
+                spawn_data_generation_stream(
+                    self.generator
+                        .into_native_stream()
+                        .map_ok(move |chunk_with_states| {
```
Comment (BugenZhao) [inline src/connector/src/source/datagen/source/reader.rs:161]: Use `inspect_ok` here?

File: src/connector/src/source/datagen/source/reader.rs [hunk 2]
```diff
@@ -151,8 +151,24 @@ impl SplitReader for DatagenSplitReader {
...
+                            metrics
+                                .partition_input_count
+                                .with_label_values(&[&actor_id, &source_id, &split_id])
+                                .inc_by(chunk_with_states.chunk.cardinality() as u64);
+                            chunk_with_states
+                        })
+                        .boxed(),
```
Comment (BugenZhao) [inline src/connector/src/source/datagen/source/reader.rs:168]: Is this `boxed` necessary here?

**Review by BugenZhao — APPROVED** [review]
Rest LGTM! Thanks for the PR.

File: src/connector/src/source/datagen/source/reader.rs (hunk 2 above)
Comment (tabVersion) [reply src/connector/src/source/datagen/source/reader.rs:168]: Looks unnecessary, will remove

**Review by waruto210 — APPROVED** [review]
LGTM

**Pull request merged**